sky/engine/core/script/dart_init.cc

```diff
@@ -341,6 +328,71 @@ void* _DartSymbolLookup(const char* symbol_name) {
   return DartLookupSymbolInLibrary(symbol_name, nullptr);
 }
 
+#elif OS(ANDROID)
+
+// Describes an asset file that holds a part of the precompiled snapshot.
+struct SymbolAsset {
+  const char* symbol_name;
+  const char* file_name;
+  bool is_executable;
+  void* mapping;
+};
+
+static SymbolAsset g_symbol_assets[] = {
+  {kDartVmIsolateSnapshotBufferName, "snapshot_aot_vmisolate", false},
+  {kDartIsolateSnapshotBufferName, "snapshot_aot_isolate", false},
+  {kInstructionsSnapshotName, "snapshot_aot_instr", true},
+  {kDataSnapshotName, "snapshot_aot_rodata", false},
+};
+
+// Resolve a precompiled snapshot symbol by mapping the corresponding asset
+// file into memory.
+void* _DartSymbolLookup(const char* symbol_name) {
+  for (SymbolAsset& symbol_asset : g_symbol_assets) {
+    if (strcmp(symbol_name, symbol_asset.symbol_name))
+      continue;
+
+    if (symbol_asset.mapping) {
+      return symbol_asset.mapping;
+    }
+
+    const std::string& aot_snapshot_path = SkySettings::Get().aot_snapshot_path;
+    CHECK(!aot_snapshot_path.empty());
+
+    base::FilePath asset_path =
+        base::FilePath(aot_snapshot_path).Append(symbol_asset.file_name);
+    int64 asset_size;
+    if (!base::GetFileSize(asset_path, &asset_size))
+      return nullptr;
+
+    int fd = open(asset_path.value().c_str(), O_RDONLY);
```

Comment:
Missing `EINTR` check. Whatever we use instead of `TEMP_FAILURE_RETRY` in GNU C.